cmd/thor/node/packer_loop.go

```diff
@@ -106,16 +106,21 @@ func (n *Node) packerLoop(ctx context.Context) {
 	}
 }
 
-func (n *Node) pack(flow *packer.Flow) error {
+func (n *Node) pack(flow *packer.Flow) (err error) {
 	txs := n.txPool.Executables()
 	var txsToRemove []*tx.Transaction
 	defer func() {
-		for _, tx := range txsToRemove {
-			n.txPool.Remove(tx.Hash(), tx.ID())
+		if err == nil {
+			for _, tx := range txsToRemove {
+				n.txPool.Remove(tx.Hash(), tx.ID())
+			}
+			metricBlockProcessedCount().AddWithLabel(1, map[string]string{"type": "proposed", "success": "true"})
+		} else {
+			metricBlockProcessedCount().AddWithLabel(1, map[string]string{"type": "proposed", "success": "false"})
 		}
 	}()
 
-	return evalBlockProposeMetrics(n.guardBlockProcessing(flow.Number(), func(conflicts uint32) error {
+	return n.guardBlockProcessing(flow.Number(), func(conflicts uint32) error {
```

Comment:
I like it, it's better than what it was, however it's not a simple flow to understand.
Just a thought, but perhaps it wouldn't be a bad idea for us to consider refactoring this in the future, worth putting a todo here?

Reply:
How about capturing it in a ticket? Seems fine to me.

Reply:
Yep, created https://github.com/vechain/protocol-board-repo/issues/209 👍